src/polyswarmd/monkey.py

```diff
@@ -16,11 +18,21 @@ def patch_gevent():
 
 def patch_web3():
     def make_post_request(endpoint_uri, data, *args, **kwargs):
+        global session
         kwargs.setdefault('timeout', 1)
-        future = session.post(endpoint_uri, data=data, *args, **kwargs)
-        response = future.result()
-        response.raise_for_status()
-
-        return response.content
+        tries = 0
+        while tries < MAX_RETRIES:
+                future = session.post(endpoint_uri, data=data, *args, **kwargs)
+            try:
+                response = future.result()
+            except ConnectionError as e:
+                tries += 1
+                logger.error('Connection error: %s', e)
+                # Create new session since the last closed
+                session = FuturesSession()
```

Comment:
session manages a thread pool where requests are issued from, concerned about the effects of dropping existing futuresession from one greenlet on the others

#164 had bumped max_retries but I removed it, think we should try that if indeed related to https://github.com/requests/requests/issues/4664 ref #169